src/dev/build/tasks/create_package_json_task.ts

```diff
@@ -37,14 +38,40 @@ export const CreatePackageJson: Task = {
 
   async run(config, log, build) {
     const pkg = config.getOpenSearchDashboardsPkg();
+    /**
+     * OpenSearch server uses the `branch` property from `package.json` to
+     * build links to the documentation. If set to `main`, it would use `/latest`
+     * and if not, it would use the `version` to construct URLs.
+     */
+    const buildVersion = config.getBuildVersion();
+    let branch;
+    if (pkg.branch === 'main') {
+      branch = pkg.branch;
+    } else {
+      const parsedBuildVersion = parse(buildVersion);
+      if (parsedBuildVersion) {
+        branch = `${parsedBuildVersion.major}.${parsedBuildVersion.minor}`;
+        log.info(`Updating package.branch to ${branch}`);
+      } else {
+        const validDocPathsPattern = /^\d+\.\d+$/;
+        if (validDocPathsPattern.test(pkg.branch as string)) {
+          branch = pkg.branch;
+        } else {
+          // package version was not parsable and branch is unusable
+          throw new Error(
```

Comment:
Does this prevent the builds if they pass a non-valid doc paths pattern in the branch in the package?

Reply:
Yes, if no version can be specified after so many downgrade logic, we will prevent the build because there will be a doc link error if we continue to do that.